src/libServer/EthRpcMethods.cpp

```diff
@@ -1748,12 +1734,43 @@ Json::Value EthRpcMethods::GetDSLeaderTxnPool() {
 }
 
 Json::Value EthRpcMethods::DebugTraceTransaction(
-    const std::string & /*txHash*/) {
-  INC_CALLS(GetInvocationsCounter());
+    const std::string& txHash, const Json::Value& json) {
 
-  if (!LOOKUP_NODE_MODE) {
-    throw JsonRpcException(ServerBase::RPC_INVALID_REQUEST,
-                           "Sent to a non-lookup");
+  if (!json.isMember("tracer")) {
+    LOG_GENERAL(WARNING, "Missing tracer field");
+    throw JsonRpcException(ServerBase::RPC_MISC_ERROR, "Missing tracer field");
+  } else {
+    auto tracer = json["tracer"].asString();
+
+    if(tracer.compare("callTracer") != 0) {
+      throw JsonRpcException(ServerBase::RPC_MISC_ERROR, std::string("Only callTracer is supported. Received: ") + tracer);
+    }
   }
-  return Json::nullValue;
+
+  std::string trace;
+  LOG_GENERAL(INFO, "Trace request: " << txHash);
+
+  try {
+    TxnHash tranHash(txHash);
+
+    bool isPresent =
+        BlockStorage::GetBlockStorage().GetTxTrace(tranHash, trace);
+
+    if (!isPresent) {
+      LOG_GENERAL(INFO, "Trace request failed! ");
+      return Json::nullValue;
+    }
+
+    Json::Value trace_json;
+    JSONUtils::GetInstance().convertStrtoJson(trace, trace_json);
+    auto const item = trace_json["call_stack"][0];
```

Comment:
The saved trace is JSON and has the map {"call_stack": [...]} - this makes it easier to later add other tracers if we desire (trace with full stack and memory, opcodes etc.)